drivers/storage/portworx/util/util.go

```diff
@@ -657,38 +662,41 @@ func AppendTLSEnv(clusterSpec *corev1.StorageClusterSpec, envMap map[string]*v1.
 func GetOciMonArgumentsForTLS(cluster *corev1.StorageCluster) ([]string, error) {
 	if cluster.Spec.Security != nil && cluster.Spec.Security.TLS != nil && cluster.Spec.Security.TLS.AdvancedTLSOptions != nil {
 		advancedOptions := cluster.Spec.Security.TLS.AdvancedTLSOptions
-		if IsEmptyOrNilCertLocation(advancedOptions.RootCA) {
-			return nil, fmt.Errorf("spec.security.tls.advancedOptions.rootCA is required")
-		}
 		if IsEmptyOrNilCertLocation(advancedOptions.ServerCert) {
 			return nil, fmt.Errorf("spec.security.tls.advancedOptions.serverCert is required")
 		}
 		if IsEmptyOrNilCertLocation(advancedOptions.ServerKey) {
 			return nil, fmt.Errorf("spec.security.tls.advancedOptions.serverKey is required")
 		}
 
-		apirootca, apicert, apikey := "", "", ""
-		if !IsEmptyOrNilSecretReference(advancedOptions.RootCA.SecretRef) {
-			apirootca = path.Join(DefaultTLSCACertMountPath, *advancedOptions.RootCA.SecretRef.SecretKey)
-		} else {
-			apirootca = path.Join(DefaultTLSCertsFolder, *advancedOptions.RootCA.FileName)
-		}
+		apicert, apikey := "", ""
 		if !IsEmptyOrNilSecretReference(advancedOptions.ServerCert.SecretRef) {
-			apicert = path.Join(DefaultTLSServerCertMountPath, *advancedOptions.ServerCert.SecretRef.SecretKey)
+			apicert = path.Join(DefaultTLSServerCertMountPath, advancedOptions.ServerCert.SecretRef.SecretKey)
 		} else {
 			apicert = path.Join(DefaultTLSCertsFolder, *advancedOptions.ServerCert.FileName)
 		}
 		if !IsEmptyOrNilSecretReference(advancedOptions.ServerKey.SecretRef) {
-			apikey = path.Join(DefaultTLSServerKeyMountPath, *advancedOptions.ServerKey.SecretRef.SecretKey)
+			apikey = path.Join(DefaultTLSServerKeyMountPath, advancedOptions.ServerKey.SecretRef.SecretKey)
 		} else {
 			apikey = path.Join(DefaultTLSCertsFolder, *advancedOptions.ServerKey.FileName)
 		}
-		return []string{
-			"-apirootca", apirootca,
+
+		args := []string{
 			"-apicert", apicert,
 			"-apikey", apikey,
 			"-apidisclientauth",
-		}, nil
+		}
+		if !IsEmptyOrNilCertLocation(advancedOptions.RootCA) {
+			if !IsEmptyOrNilSecretReference(advancedOptions.RootCA.SecretRef) {
+				args = append(args, "-apirootca", path.Join(DefaultTLSCACertMountPath, advancedOptions.RootCA.SecretRef.SecretKey))
+			} else if !IsEmptyOrNilStringPtr(advancedOptions.RootCA.FileName) {
+				args = append(args, "-apirootca", path.Join(DefaultTLSCertsFolder, *advancedOptions.RootCA.FileName))
```

Comment:
This still assumes the default base path - will this take a direct full path in a future PR? (as per our discussion today)

Reply:
> This still assumes the default base path - will this take a direct full path in a future PR? (as per our discussion today)

Yes, there was a bit of feedback from the meeting today. I'm going to create a JIRA (for this and possibly other changes). I'll validate with other stakeholders to get an agreement from them.

This is still in the feature branch, so we should be ok for now